src/dune_rules/melange/melange_rules.ml

```diff
@@ -216,27 +216,32 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
     let stdlib_dir = ctx.stdlib_dir in
     let+ () =
       let target_dir = Path.Build.relative dir mel.target in
-      match mel.alias with
-      | None -> Memo.return ()
-      | Some alias_name ->
-        let module_systems = mel.module_systems in
-        let deps =
-          js_targets_of_modules ~output:(`Private_library_or_emit target_dir)
-            ~module_systems modules
-          |> Action_builder.path_set
-        in
-        let alias = Alias.make alias_name ~dir in
-        let* () = Rules.Produce.Alias.add_deps alias deps in
-        (let open Action_builder.O in
+      let module_systems = mel.module_systems in
+      let emit_deps =
+        js_targets_of_modules ~output:(`Private_library_or_emit target_dir)
+          ~module_systems modules
+        |> Action_builder.path_set
+      in
+      let lib_deps =
+        let open Action_builder.O in
         let* deps =
           Resolve.Memo.read
           @@
           let open Resolve.Memo.O in
           Compilation_context.requires_link cctx
           >>= js_targets_of_libs sctx ~module_systems ~target_dir
         in
-        Action_builder.paths deps)
-        |> Rules.Produce.Alias.add_deps alias
+        Action_builder.paths deps
+      in
+      let alias = Alias.make Melange_stanzas.Emit.implicit_alias ~dir in
+      let* () = Rules.Produce.Alias.add_deps alias emit_deps in
+      let* () = Rules.Produce.Alias.add_deps alias lib_deps in
```

Comment:
```suggestion
      let* () = Rules.Produce.Alias.add_deps alias (Action_builder.both emit_deps lib_deps) in
```

Reply:
`add_deps` expects `unit Action_builder.t` so this change doesn't work.

```
This expression has type (unit * unit) Action_builder.t
but an expression was expected of type unit Action_builder.t
```

Reply:
you can toss an `Action_builder.ignore` or use the let syntax.

Reply:
You mean st like this?

```
let* () =
        Rules.Produce.Alias.add_deps alias
          (Action_builder.ignore (Action_builder.both emit_deps lib_deps))
```

Sorry, I think I still don't understand what you mean, and never feel comfortable using `ignore`.

Reply:
Please let me know if https://github.com/ocaml/dune/pull/7327/commits/f9c061e6ea1fe2084d06ba97c4767a7d7a73ac0c is what you had in mind.